data/service/api/v1/users_datasets_create.go

```diff
@@ -69,7 +69,9 @@ func UsersDataSetsCreate(dataServiceContext dataService.Context) {
 	}
 
 	dataSet.SetUserID(&targetUserID)
-	dataSet.SetCreatedUserID(pointer.FromString(details.UserID()))
+	if details.IsUser() {
+		dataSet.SetCreatedUserID(pointer.FromString(details.UserID()))
```

Comment:
FYI, `details.UserID` will be set to the id backend client service account, which is a real user in Keycloak.

Reply:
Hmm, that's not what I'm seeing. I'm getting an empty string which causes a failure during later validation (`CreatedUserID` of `nil` is fine, but an empty string is not).

Reply:
I'll double check.

Reply:
While the service session token does include the `UserID` of the backend client service account, it is not being used because the session token is marked as `IsServer`. Find the relevant code that ignores the `UserID` for a server session token at https://github.com/tidepool-org/platform/blob/f7976305c9b2324d441c82a45f051e03ce0ff95d/auth/client/external.go#L208.